File: examples/wasm/hello.kt
// FunctionFly example: Hello from Kotlin
// Build: gradle wasmWasiNodeProductionRun
// Manifest: {"name":"hello-kotlin","version":"1.0.0","runtime":"kotlin","entry":"Main.kt"}

fun main() {
    val input = generateSequence { readlnOrNull() }.joinToString("\n")
    val response = """{"message": "Hello from Kotlin!", "input_length": ${input.length}, "ok": true}"""
    print(response)
}

File: examples/wasm/hello.swift
// FunctionFly example: Hello from Swift
// Build: swiftc -target wasm32-unknown-wasi hello.swift -o hello.wasm
// Manifest: {"name":"hello-swift","version":"1.0.0","runtime":"swift","entry":"hello.swift"}
import Foundation

var input = ""
while let line = readLine() {
    input += line + "\n"
}

let response = #"{"message": "Hello from Swift!", "input_length": \#(input.count), "ok": true}"#
print(response, terminator: "")
